### PR TITLE
bump(*)

### PR DESCRIPTION
**Pull request by bparees**
the point of this PR is to update glide.yaml to use the new image-registry-release-2.6.2 branch in openshift/docker-distribution.

everything else is fall out of running glide update after making that change....  so it needs validation.  In particular we should not have seen any changes to the docker-distribution vendor dir itself, and yet there is one, so we need to understand where that came from.

update: that has been resolved by merging the carry into the fork.

### CODE REVIEW

File: vendor/github.com/docker/distribution/configuration/parser.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func (p *Parser) overwriteStruct(v reflect.Value, fullpath string, path []string
 		}
 	case reflect.Ptr:
 		if field.IsNil() {
-			field.Set(reflect.New(field.Type().Elem()))
+			field.Set(reflect.New(sf.Type))
```
Comment (bparees) [inline vendor/github.com/docker/distribution/configuration/parser.go]: i'm not sure why this change is being picked up unless we were carrying a patch in our vendor dir that is not in openshift/docker-distribution ?

**Comment by deads2k** [comment]
@legionus @dmage @miminar did you guys find a way to eliminate `UPSTREAM: docker/distribution: <drop>: pull vendor deps` now that you're no longer in a repo trying to build `kubelet`?  That would be great, but I didn't notice it go by so I may have given @bparees out of date advice.

File: vendor/github.com/docker/distribution/configuration/parser.go (hunk 1 above)
Comment (bparees) [reply vendor/github.com/docker/distribution/configuration/parser.go]: it looks like the change came from 7746110b40a1689628d79c8ba35822746ffd3160 

so we need to get 7746110b40a1689628d79c8ba35822746ffd3160 into openshift/docker-distribution#image-registry-release-2.6.2

Comment (bparees) [reply vendor/github.com/docker/distribution/configuration/parser.go]: pr here: https://github.com/openshift/docker-distribution/pull/2

**Comment by dmage** [comment]
@deads2k yes, `glide up -v` and one patch for distribution should be enough.

**Comment by bparees** [comment]
thanks @dmage, i dropped the carry.  after the change merges into docker-distribution i'll rerun glide up on this PR.

File: vendor/github.com/docker/distribution/configuration/parser.go (hunk 1 above)
Comment (legionus) [reply vendor/github.com/docker/distribution/configuration/parser.go]: > i'm not sure why this change is being picked up unless we were carrying a patch in our vendor dir that is not in openshift/docker-distribution ?

@mfojtik told me that changes in the `vendor` directory should be made in our repository and then transferred to the `openshift/docker-distribution` repository by some script.

Before that, I was opening PR: https://github.com/openshift/docker-distribution/pull/1

Comment (bparees) [reply vendor/github.com/docker/distribution/configuration/parser.go]: well barring more details on who's implementing that script, i'm going to proceed with the manual approach.

**Comment by bparees** [comment]
ok i think this is finally doing what i'd expect.  Our docker-distribution vendor dep now comes out of the openshift/docker-distribution:image-registry-release-2.6.2 fork+branch.

Going forward, we should submit carries to that fork/branch and then vendor back to here.

@dmage @legionus @deads2k ptal

**Comment by legionus** [comment]
>  Our docker-distribution vendor dep now comes out of the openshift/docker-distribution:image-registry-release-2.6.2 fork+branch.

@bparees Only you will have permissions to merge there ?

**Comment by bparees** [comment]
> @bparees Only you will have permissions to merge there ?

any openshift project committer (there are about 22 last i knew).  Certainly we can consider expanding that as well, or perhaps we'll get the tooling that @mfojtik alluded to.
